src/main/scala/pl/project13/kaze/KazeClass.scala

```diff
@@ -29,54 +31,117 @@ object KazeClass {
       classOf[immutable.Seq[_]].getName, classOf[Map[_, _]].getName)
 
     private val fields = clazz.getDeclaredFields
-        .filterNot(skip)
-        .filterNot(_.getName.contains("$"))
+      .filterNot(skip)
+      .filterNot(_.getName.contains("$"))
 
     def render = {
       val sb = new StringBuilder(s"${indent}final class $clazzName private(\n")
       indent = "  "
 
-      // constructor
+      renderConstructor(sb)
+
+      sb.append("{\n\n")
+
+      renderJavaAccessors(sb)
+
+      sb.append("\n")
+
+      renderWithXyz(sb)
+      sb.append("\n")
+      renderCopy(sb)
+      sb.append("\n")
+      renderToString(sb)
+      sb.append("\n")
+      renderEquals(sb)
+      sb.append("\n")
+      renderHashCode(sb)
+
+      sb.append("}\n") // end class
+
+      // companion object
+      indent = "  "
+      sb.append(s"object $clazzName {\n")
+      sb.append(s"${indent}/** Scala API */\n")
+      sb.append(s"${indent}def apply(): $clazzName = new $clazzName()\n")
+
+      sb.append(s"${indent}/** Java API */\n")
+      sb.append(s"${indent}def getInstance(): $clazzName = apply()\n")
```

Comment:
Typically those factories should be from a `Config` parameter and `ActorSystem` parameter that grabs the `Config`. It might be too much to try to generate all that, but then `getInstance` is not a good name. `create` would be better.

Example https://github.com/akka/akka/blob/master/akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala#L57

Reply:
> Typically those factories should be from a Config parameter and ActorSystem parameter that grabs the Config.

Yeah good point... could you do a follow up PR @ennru ?


> create would be better.

+1, mentioned before.

Reply:
The parameter-less is just like before, but create sounds better.

For the all-fields variant, I have the need to get it prepared so that I can manually remove parameters that are not required, just leaving the ones that don't have good defaults.

Generation config reader would be great, but a different story.

Reply:
defaults should be defined in config, so there are no good defaults apart from `Option` fields